File: app/src/main/java/com/example/pizzaHut/utilities/BindingAdapters.java
package com.example.pizzaHut.utilities;

import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.databinding.BindingAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

public class BindingAdapters {

    @BindingAdapter("android:imageURL")
    public static void setImageURL(ImageView imageView, String URL) {
        try {
            imageView.setAlpha(0f);
            Picasso.get().load(URL).noFade().into(imageView, new Callback() {
                @Override
                public void onSuccess() {
                    imageView.animate().setDuration(300).alpha(1f).start();
                }

                @Override
                public void onError(Exception e) {

                }
            });
        } catch (Exception ignored) {

        }
    }

    @BindingAdapter("android:trim")
    public static void setTrim(TextView textView, String text) {
        try {
            textView.setText(text.trim());
        } catch (Exception ignored) {

        }
    }
}

File: app/src/main/java/com/example/foodRecipes/util/BindingAdapters.kt
package com.example.foodRecipes.util

import android.content.Intent
import android.net.Uri
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.widget.AppCompatButton
import androidx.databinding.BindingAdapter
import coil.load

@BindingAdapter("android:imageURL")
fun setImageURL(imageView: ImageView, URL: String?) = try {
    imageView.load(URL)
} catch (ignored: Exception) {
}

@BindingAdapter("android:trim")
fun setTrim(textView: TextView?, text: String?) = try {
    textView?.text = text?.trim { it <= ' ' }
} catch (ignored: Exception) {
}

@BindingAdapter("android:buttonURL")
fun initClickListeners(button: AppCompatButton, link: String?) {
    button.setOnClickListener {
        if (link == null)
            Toast.makeText(button.context, "Link is NULL", Toast.LENGTH_SHORT).show()
        else button.context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(link)))
    }
}